kvclient: let QueryLocks and LeaseInfo requests through txnWriteBuffer

yuzefovich opened the pull request:
These requests don't interact with buffered writes, so we simply allow the requests and their response to go through the interceptor unchanged.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/144276)
<!-- Reviewable:end -->

arulajmani reviewed (commented):
I agree with you, not having any special logic here should be good. In fact, I think that's an improvement to the current behaviour (without buffered writes) because of the way the `QueryLockRequest` works -- it doesn't include uncontended replicated locks (intents). 

With buffered writes, before the flush, we'll see any unreplicated `Exclusive` locks we've acquired because of the decomposition that we do. That's better than flushing, because:
- Flushing will needlessly increase the contention profile (we'll have intents (which act as locks) on non-unique secondary indexes)
- If the flushed intents are uncontended, we'll not see them in the `QueryLockResponse`. This is a quirk of `QueryLockRequest` which isn't great to begin with.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144276#-:-ON_9nhz569JAnhOZBxk:bvzepzx)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @michae2)



<!-- Sent from Reviewable.io -->

yuzefovich committed:
kvclient: let QueryLocks and LeaseInfo requests through txnWriteBuffer

These requests don't interact with buffered writes, so we simply allow
the requests and their response to go through the interceptor unchanged.

Release note: None

yuzefovich commented:
I updated the PR to let these two requests through the interceptor unchanged (I'm assuming that's what you were suggesting). PTAL.

yuzefovich commented:
btw the updated `cluster_locks` and `cluster_locks_tenant` files with buffered writes enabled are [here](https://github.com/cockroachdb/cockroach/pull/143860/commits/03dc159107d43fc243f0f172d4e7e585d07befaa)

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144276#-:-ON_ukA-4EF3bq1nFrK5:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @michae2)



<!-- Sent from Reviewable.io -->

arulajmani commented:
Yup, that's what I was suggesting. 

bors r+

yuzefovich commented:
Seems like an unrelated flake.

bors r+

yuzefovich commented:
blathers backport release-25.2

yuzefovich commented:
blathers backport release-25.2
